unitTests/ui/InitializationTest.ts

```diff
@@ -150,30 +150,33 @@ export function InitializationTest() {
       });
 
       expect(Component.get(queryBox) instanceof Querybox).toBe(false);
-      Initialization.initializeFramework(root, searchInterfaceOptions, () => {
+      await Initialization.initializeFramework(root, searchInterfaceOptions, () => {
         return Initialization.initSearchInterface(root, searchInterfaceOptions);
       });
       expect(Component.get(queryBox) instanceof Querybox).toBe(true);
       const sBox = <Querybox>Component.get(queryBox);
       expect(sBox.options.enableSearchAsYouType).toBe(true);
       expect(sBox.options.enablePartialMatch).toBe(true);
+      done();
     });
 
-    it('allows to registerAutoCreateComponent', () => {
+    it('allows to registerAutoCreateComponent', async done => {
       const dummyCmp: any = jasmine.createSpy('foobar');
       dummyCmp.ID = 'FooBar';
       const dummyElem = document.createElement('div');
       $$(dummyElem).addClass('CoveoFooBar');
       root.appendChild(dummyElem);
 
       Initialization.registerAutoCreateComponent(dummyCmp);
-      Initialization.initializeFramework(root, searchInterfaceOptions, () => {
+      await Initialization.initializeFramework(root, searchInterfaceOptions, () => {
```

Comment:
Here, it's kind of a clear sign that the method signature has changed, and this could break a lot of pages.

In non-lazy init mode, everything should be initialized after the function has finished executing, and we should not have to modify the test to `await` the initialization.

This goes for this whole test file.